### PR TITLE
Calibration feature for Octoliner.

### LINKED ISSUE #6: Backport calibration feature from Arduino library
The 2.1.0 version of Arduino Octoliner Library introduces a method to automatically select sensitivity. See: https://github.com/amperka/Octoliner/commit/fdf1682391899f9fb501f6e2554c49219673b9ad

The same functionality is much desired in OctolinerPi

### PR DESCRIPTION
**Pull request by AlexErAmp**
Close #6

### CODE REVIEW

File: examples/autocalibrate.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                'Please enter the command "calc" to calculate the sensitivity',
+                'or Ctrl^C to exit.'
+            )
+            command = input()
+            if command == "calc":
+                # Optimal sensitivity calculation.
+                octo.optimize_sensitivity_on_black()
+                print("Optimal sensitivity: %.2f\n" % octo.get_sensitivity())
+            else:
+                print("Command is invalid")
```
Comment (nkrkv) [inline examples/autocalibrate.py]: The “calc” command is somewhat complicated. Could we accept just Enter press and not consider the actual input at all?

Comment (AlexErAmp) [reply examples/autocalibrate.py]: Done!

**Review by nkrkv — APPROVED** [review]
LGTM

File: octoliner/octoliner.py [hunk 2]
```diff
@@ -80,7 +102,82 @@ def set_sensitivity(self, sense):
...
+        while sens < 1.0:
+            self.set_sensitivity(sens)
+            time.sleep(0.05)
+            if self._count_of_black() != 8:
+                break
+            # Choose a more accurate step value as 1.0 / 250.
+            sens += 0.004
+
+        # Environment has changed since the start of the process.
+        if sens == 1.0:
```
Comment (yurybotov) [inline octoliner/octoliner.py]: Here must be "if sens >= 1.0:"

Comment (AlexErAmp) [reply octoliner/octoliner.py]: Done!

**Review by yurybotov — APPROVED** [review]
Ok

**Pull request merged**